api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -96,31 +97,42 @@ public Workspace get(String ns, String firecloudName) {
     List<Workspace> dbWorkspaces = workspaceDao.findAllByFirecloudUuidIn(fcWorkspaces.keySet());
 
     return dbWorkspaces.stream()
+        .filter(dbWorkspace -> dbWorkspace.getWorkspaceActiveStatusEnum() == WorkspaceActiveStatus.ACTIVE)
         .map(dbWorkspace -> {
           String fcWorkspaceAccessLevel = fcWorkspaces.get(dbWorkspace.getFirecloudUuid()).getAccessLevel();
           WorkspaceResponse currentWorkspace = new WorkspaceResponse();
           currentWorkspace.setWorkspace(workspaceMapper.toApiWorkspace(dbWorkspace));
           currentWorkspace.setAccessLevel(workspaceMapper.toApiWorkspaceAccessLevel(fcWorkspaceAccessLevel));
           return currentWorkspace;
-        }).collect(Collectors.toList());
+        })
+        .collect(Collectors.toList());
   }
 
   private Map<String, org.pmiops.workbench.firecloud.model.WorkspaceResponse> getFirecloudWorkspaces() {
-    return fireCloudService.getWorkspaces().stream().collect(
-            Collectors.toMap(
+    return fireCloudService.getWorkspaces().stream()
+            .collect(Collectors.toMap(
                 fcWorkspace -> fcWorkspace.getWorkspace().getWorkspaceId(),
                 fcWorkspace -> fcWorkspace));
   }
 
+  /**
+   * This is an internal method used by createWorkspace and cloneWorkspace endpoints, to
+   * check the existance of ws name.  Currently does not return a conflict if user
+   * is checking the name of a deleted ws.
+   **/
   @Override
   public Workspace getByName(String ns, String name) {
-    return workspaceDao.findByWorkspaceNamespaceAndName(ns, name);
+    Workspace workspace = workspaceDao.findByWorkspaceNamespaceAndName(ns, name);
+    if (workspace == null || workspace.getWorkspaceActiveStatusEnum() != WorkspaceActiveStatus.ACTIVE) {
+      return null;
+    }
+    return workspace;
   }
 
   @Override
   public Workspace getRequired(String ns, String firecloudName) {
     Workspace workspace = get(ns, firecloudName);
-    if (workspace == null) {
+    if (workspace == null || (workspace.getWorkspaceActiveStatusEnum() != WorkspaceActiveStatus.ACTIVE)) {
```

Comment:
Therea re a few references to `enforceWorkspaceAccessLevel` which I think should be covered, e.g. copyNotebook().

Reply:
what do you mean? like a user should not be able to copy a notebook from a deleted workspace?

Reply:
Yes - and it should be 404ing if I try that. Perhaps it already works since we're checking FC for the ACL, but worth verifying.

There are several cohort methods that also use this method as their only access check.